protocol/pubsub/v2/message_test.go

```diff
@@ -45,3 +46,39 @@ func TestReadStructured(t *testing.T) {
 		})
 	}
 }
+
+func TestFinish(t *testing.T) {
+	tests := []struct {
+		name    string
+		pm      *pubsub.Message
+		err     error
+		wantErr bool
+	}{
+		{
+			name: "return error",
+			pm: &pubsub.Message{
+				ID: "testid",
+			},
+			err:     fmt.Errorf("error"),
+			wantErr: true,
+		},
+		{
+			name: "no errors",
+			pm: &pubsub.Message{
+				ID: "testid",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			msg := NewMessage(tc.pm)
+			err := msg.Finish(tc.err)
+			if tc.wantErr {
+				if err != tc.err {
+					t.Errorf("Error mismatch. got: %v, want: %v", err, tc.err)
+				}
+			}
```

Comment:
nit: logic is partially broken IMHO if `tc.wantErr` is `false` but`err != nil`, i.e., the second test case does not seem to have any effect here?

Usually you'd do something like

```suggestion
			if tc.wantErr && err != nil {
				if err != tc.err {
					t.Errorf("Error mismatch. got: %v, want: %v", err, tc.err)
				}
			} else { // assert `tc.wantErr` is `false` and `err == nil` }
```

Reply:
yep, makes sense

Reply:
done, thanks for the review